flexdll.c

```diff
@@ -496,8 +498,22 @@ void *flexdll_wdlopen(const wchar_t *file, int mode) {
 #endif /* __STDC_SECURE_LIB__ >= 200411L*/
 #endif /* CYGWIN */
 
+again:
+  if (units_mutex == INVALID_HANDLE_VALUE) {
+    HANDLE hMutex = CreateMutex(NULL, TRUE, NULL);
+    if (InterlockedCompareExchangePointer(&units_mutex, hMutex, INVALID_HANDLE_VALUE) != INVALID_HANDLE_VALUE) {
+      CloseHandle(hMutex);
+      goto again;
+    }
+  } else {
+    if (WaitForSingleObject(units_mutex, INFINITE) == WAIT_FAILED) {
```

Comment:
```suggestion
    if (WaitForSingleObject(units_mutex, INFINITE) != WAIT_OBJECT_0) {
```
This would cover the improbable (impossible?) case of the mutex having been abandoned.
https://learn.microsoft.com/en-us/windows/win32/sync/using-mutex-objects

Reply:
I ended up reverting this one.

`WAIT_ABANDONED` does not offer `GetLastError`-integration, so I started adding an explicit error message case for it - only to then hit it during testing.

I therefore suppose these two cases were written to handle both `WAIT_OBJECT_0` and `WAIT_ABANDONED` as success-cases (`WAIT_TIMEOUT` should not happen with `INFINITE`...)

Reply:
My reasoning I think is that WAIT_ABANDONED should be treated like success?